068-access-policies-in-jwt.md

```diff
@@ -78,6 +78,41 @@ and for a string:
 
 The field name is configurable with default value `acls`.
 
+### ACL Syntax
+
+A single ACL has the syntax `CLUSTER_NAME:RESOURCE_TYPE:RESOURCE_SPEC:PERMITTED_ACTIONS`.
+The separator `:` is chosen because it is a common separator in the Unix world.
+
+If the ACLs are passed as a string, they are separated by `,`.
+i.e., `CLUSTER_NAME:RESOURCE_TYPE:RESOURCE_SPEC:PERMITTED_ACTIONS,CLUSTER_NAME:RESOURCE_TYPE:RESOURCE_SPEC:PERMITTED_ACTIONS`
+
+- `RESOURCE_TYPE` can be `topic` or `group`, or the shortened versions `t` and `g`.
+- `PERMITTED_ACTIONS` is a list of a subset of `read`, `write`, `create`, `delete`, `alter`, `alter_configs`, `describe`, `describe_configs`, `cluster_action`, `all`, or the shortened versions `r`, `w`, `c`, `d`, `a`, `ac`, `de`, `dc`, `ca`. The short version for `all` is `*`.
+The list items are separated by `|`.
+The choice of the separator `|` is based on the typical pattern of OR-ing boolean permissions.
+- `CLUSTER_NAME` is the name of the cluster, and `RESOURCE_SPEC` is the name of the resource (topic/group).
+These fields can start or end with `*` to match any prefix/suffix.
+
+### Defaults
+If `CLUSTER_NAME` or `RESOURCE_SPEC` is empty, it is considered a wildcard `*`, which matches any cluster or resource.
+
+If `RESOURCE_TYPE` is empty, it takes the default value `topic`.
+
+If `PERMITTED_ACTIONS` is empty, no actions are allowed.
+
+The defaults for `CLUSTER_NAME`, `RESOURCE_SPEC`, and `RESOURCE_TYPE` are chosen for convenience.
+The default for `PERMITTED_ACTIONS` is none to prevent accidental access to all resources.
+
+### Examples of ACLs
+
+- `my_cluster:t:topic1:r|w` - allows reading and writing to `topic1` in the cluster `my_cluster`.
```

Comment:
Why did you change from `,` to `|` for permitted actions? Now the symbol reads "or", but the description says "and", which is kind of weird.

Reply:
My intuition for `|` is the same as when talking about read/write/execute file permissions in Unix, where permissions are represented as 3 bits, so if you want to enable read and write you turn on the read and write permission, which is done by OR-ing them. `100 | 010 = 110`
Again, I have no strong preference, this is the first thing that came to my mind. I would prefer not to use a `,` for AND and instead use it to separate the different ACLs. I think the `&` sign is also a bad idea as it's not as easy to read alongside words.

Reply:
I like the `+`